docs/evaluation/tutorials/agents.mdx

```diff
@@ -1,602 +1,1989 @@
----
-sidebar_position: 6
----
-
 import { RegionalUrl } from "@site/src/components/RegionalUrls";
 
 # Evaluate an agent
 
-In this tutorial, we will walk through 3 evaluation strategies LLM agents, building on the conceptual points shared in our [evaluation guide](../concepts#agents).
+:::info Key concepts
+[Agent evaluation](../concepts#agents) | [Evaluators](../concepts#evaluators) | [LLM-as-judge evaluators](../concepts#llm-as-judge)
+:::
+
+In this tutorial, we'll build a customer support bot that helps users navigate a digital music store. We'll create three types of evaluations:
 
-- `Final Response`: Evaluate the agent's final response.
-- `Single step`: Evaluate any agent step in isolation (e.g., whether it selects the appropriate tool).
-- `Trajectory`: Evaluate whether the agent took the expected path (e.g., of tool calls) to arrive at the final answer.
+- [Final response](../concepts#evaluating-an-agents-final-response): Evaluate the agent's final response.
+- [Trajectory](../concepts#evaluating-an-agents-trajectory): Evaluate whether the agent took the expected path (e.g., of tool calls) to arrive at the final answer.
+- [Single step](../concepts#evaluating-a-single-step-of-an-agent): Evaluate any agent step in isolation (e.g., whether it selects the appropriate first tool for a given ).
 
-First, we will build an agent using [LangGraph](https://github.com/langchain-ai/langgraph).
+We'll build our agent using [LangGraph](https://github.com/langchain-ai/langgraph), but the techniques and LangSmith functionality shown here are framework-agnostic.
 
-## Set up environment
+## Setup
 
-We'll set up our environment variables for OpenAI and <RegionalUrl text='LangSmith'/> and install
+### Configure the environment
+
+Let's install the required dependencies:
 
 ```bash
-%pip install --upgrade --quiet langchain langsmith langchain-community langchain-experimental langgraph
+pip install -U langgraph langchain langchain-community langchain-openai
 ```
 
+and set up our environment variables for OpenAI and <RegionalUrl text='LangSmith'/>:
+
 ```python
+#region
 import getpass
 import os
 
-def _set_env(var: str):
+def _set_env(var: str) -> None:
     if not os.environ.get(var):
-        os.environ[var] = getpass.getpass(f"{var}: ")
+        os.environ[var] = getpass.getpass(f"Set {var}: ")
 
-_set_env("OPENAI_API_KEY")
 os.environ["LANGCHAIN_TRACING_V2"] = "true"
-os.environ["LANGCHAIN_ENDPOINT"] = "https://api.smith.langchain.com" # Update appropriately for self-hosted installations or the EU region
 _set_env("LANGCHAIN_API_KEY")
+_set_env("OPENAI_API_KEY")
+#endregion
 ```
 
```

Comment:
While we're using OpenAI for this example, you can use any [LangChain supported chat model](https://python.langchain.com/docs/integrations/chat/#all-chat-models).